Prometheus Governance 2.0

gouthamve opened the pull request:
Previous feedback has been captured [here](https://docs.google.com/document/d/1fnDfySwINOPk9OPSsV_m2Uzf1NZU9pf7QaUSiOQAT54/edit?tab=t.0)

Please provide any feedback you might have.

gouthamve committed:
Prometheus Governance 2.0

Previous feedback has been captured [here](https://docs.google.com/document/d/1fnDfySwINOPk9OPSsV_m2Uzf1NZU9pf7QaUSiOQAT54/edit?tab=t.0)

Please provide any feedback you might have.

Signed-off-by: gouthamve <gouthamve@gmail.com>

jkroepke commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+# Prometheus Governance 2.0
+
+This document describes the rules and governance of the project. It is meant to be followed by all the developers of the project and the Prometheus community. This governance is an open, living document, and will continue to evolve as the community and project change.
```
The document keeps referring to "the project" instead of directly mentioning Prometheus. It feels like Fight Club (Rule 1: Don't talk about Fight Club).

It would be helpful to add an introduction that clearly states what the project is.

beorn7 replied:
Agreed. Some things are kind-of implicit by mentioning GH orgs etc. later. But it's still not clear that this might be for more than just the code (like events, communication channels outside of GH, …).

metalmatze replied:
That's probably because we copied this sentence and some more from a CNCF template. We should update this. Good catch!

jkroepke reviewed (commented):
Hi, I'm trying to give some feedback from a member perspective that recently joined the prometheus team. Could be possible that it is crystal clear from you point of view, however for me it's a bit difficult to read.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+The existing Prometheus team is going to hold votes, as described in the new governance, to elect a Bootstrap steering committee of 7 members, where the highest voted 3 members will hold a 2 year and the next 4 members will hold a 1-year term.  
+
+
+[^1]:	[github.com/prometheus](http://github.com/prometheus), [github.com/prometheus-community](http://github.com/prometheus-community), [github.com/promcon](http://github.com/promcon), [github.com/openmetrics](http://github.com/openmetrics)
```
http://github.com/openmetrics is gone.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+It outlines the roles and responsibilities of the Steering Committee, the decision-making process, and the election procedures for committee members. It also details the contributor ladder, including the roles of Contributor, Member, and Maintainer, and the requirements and privileges associated with each role.
+
+The Prometheus Steering Committee is the governing body of the Prometheus project, providing decision-making and oversight pertaining to the project bylaws, sub-organizations[^1], and financial planning. The Steering Committee also defines the project values and structure.
```
Why "sub-organizations"? With the footnote, I would just say "GitHub organizations".

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Roles
+
+We have a contributor ladder that includes the roles of Contributor, Member, Maintainer and Steering Committee. The steering committee responsibilities and elections are outlined below. The rest of the roles are detailed in [ROLES.md](#roles.md).
```
Oxford comma is used further down. We should keep things consistent. I would use it everywhere (as we generally do in Prometheus docs).

```suggestion
We have a contributor ladder that includes the roles of Contributor, Member, Maintainer, and Steering Committee. The steering committee responsibilities and elections are outlined below. The rest of the roles are detailed in [ROLES.md](#roles.md).
```

RichiH replied:
```suggestion
The project has a contributor ladder that includes the roles of Contributor, Member, Maintainer and Steering Committee. The steering committee responsibilities and elections are outlined below. Additional roles are detailed in [ROLES.md](#roles.md).
```

beorn7 replied:
Note that `#roles.md` will not point to the file `roles.md`. I guess you meant to not use `#` here.

siavashs replied:
```suggestion
We have a contributor ladder that includes the roles of Contributor, Member, Maintainer and Steering Committee. The steering committee responsibilities and elections are outlined below. The rest of the roles are detailed in [ROLES.md](./ROLES.md).
```
Also opened #3

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+The following responsibilities and powers belong to the Steering Committee.
+
+* Define, evolve, and promote the vision, values, and mission of the project.
+* Define and evolve project governance structures and policies, including project roles and how collaborators become members, approvers, leads, and/or administrators. This includes policy for the creation and administration of working groups and committees.
```
"members, approvers, leads, and/or administrators" – this is an awfully specific list of roles, but those roles do not formally exist right now (except "members"). Is this an oversight from a previous version of this text? Maybe this:

```suggestion
* Define and evolve project governance structures and policies, including project roles and how collaborators adopt these roles. This includes policy for the creation and administration of working groups and committees.
```

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+* Coordinate with the CNCF regarding usage of the Prometheus brand and how that brand can be used in relation to other efforts or vendors.
+* Receive and handle reports about code of conduct violations and maintain confidentiality.
+* Act as the final escalation point and decide for any disputes, issues, clarifications, or escalations within the project scope.
+* Have “owners” role for the `prometheus` and `prometheus-community` github organizations.
```
Shouldn't these be all GH organizations, as in the footnote? (Just footnote it again here?)

```suggestion
* Have “owners” role for the project's GitHub organizations[^1].
```

(Also note inconsistent spelling of "GitHub".)

RichiH replied:
We can likely remove this after the cleanup; but we can also leave it in as we will likely not delete the second org.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+* Receive and handle reports about code of conduct violations and maintain confidentiality.
+* Act as the final escalation point and decide for any disputes, issues, clarifications, or escalations within the project scope.
+* Have “owners” role for the `prometheus` and `prometheus-community` github organizations.
+* Ultimate responsibility for all key project functions, unless delegated to a team. For example, vulnerability and security report handling, or github administration.
```
github → GitHub
(Or other spelling, but keep it consistent throughout the doc.)

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+
+* Responsibilities include
+  * Continues to contribute regularly.
+	* This can be demonstrated by having at least 20 Github contributions a year in [DevStats](https://devstats.cncf.io/).
```
Github → GitHub
(Or other spelling, but keep it consistent throughout the doc.)

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+	* Organizing and running activities (e.g. events, surveys) within the Prometheus community
+	* Contributing to SIGs, subprojects, or community discussions (e.g. meetings, chat, email, and discussion forums)
+	* Must have two or more Maintainers as sponsors. At least one sponsor must have a different employer than the applying Contributor
+	* Must enable 2FA on their Github account
```
Github → GitHub
(Or other spelling, but keep it consistent throughout the doc.)

beorn7 replied:
These two should be outdented to the 2nd level (only two leading blanks).

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+  * May be assigned Issues and Reviews
+  * May give commands to CI/CD automation
+  * Can be added to repository teams
+  * Added as a member of the Github org
```
Github → GitHub
(Or other spelling, but keep it consistent throughout the doc.)

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+The Steering Committee defaults to deciding by consensus.
+
+Decisions requiring a vote include: issuing written policy, amending existing  
+written policy, creating, removing, or modifying a working group, all spending,  
+hiring, and contracting, official responses to publicly raised issues, or any  
+other decisions that at least two of the members present decide require a vote.
```
Grouping is not clear here. Could use `;` for the second-level grouping, or maybe this:

```suggestion
Decisions requiring a vote include:
* Issuing written policy
* Amending existing written policy
* Creating, removing, or modifying a working group
* All spending, hiring, and contracting
* Official responses to publicly raised issues
* Any other decisions that at least two of the members present decide to require a vote
```

(I also added a "to" in the last item there. Those knowing English better than me should verify.)

sysadmind replied:
I like the list idea that Björn suggested. I think to the point on the final item, it would be better to reword it a bit. My suggestion: `Any other decisions that two or more members present agree should be put to a vote`

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+hiring, and contracting, official responses to publicly raised issues, or any  
+other decisions that at least two of the members present decide require a vote.
+
+Votes happen by simple majority.
```
Is that a sufficient definition?

How to count abstains? How to count members not present or not taking part for other reasons?

Given that there are only 7 seats, should we simply require 4 yes votes?

Do we need to rule out multi-way votes? (If it's not a yes/no vote but "which of these 6 proposals do we want to accept"?) If we don't, what's a "simple majority" then? Two members vote for proposal 1, all other 5 proposals get one vote, for example…

I would like to keep it really simple for starters, e.g.

```suggestion
To pass a decision, it needs approving votes from more than half of the current steering committee members.
```

sysadmind replied:
I agree with the feedback here. If we are taking the time to spell out the governance, I think we should put in the time to spell out the voting. 

To the point about potential votes that are not yes/no, what if this suggested that all votes be phrased as yes/no? Yes/no voting keeps this document simple. There could be scenarios that do not make sense to phrase as a yes/no, but if votes **should** be yes/no, I think that may cover the most likely scenarios. In the proposal case above, the vote could be 6 yes/no votes.

RichiH replied:
Quorums are usually distinct from voting results; this would also make sense here.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+There are two ways to raise issues to the steering committee for decision:
+
+1. The Steering Committee can be reached privately by emailing [prometheus-steering@googlegroups.com](mailto:prometheus-steering@googlegroups.com). This is a private discussion list to which all members of the committee have access.
```
Do we really want a Google group for this? Can we maintain a mail alias within prometheus.io? It's only for seven people, it should be fine. Then we could use `steering@prometheus.io`. It would also remove the problem of having a separate archive with loads of confidential information that we maybe need to delete when the committee members rotate.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+There are two ways to raise issues to the steering committee for decision:
+
+1. The Steering Committee can be reached privately by emailing [prometheus-steering@googlegroups.com](mailto:prometheus-steering@googlegroups.com). This is a private discussion list to which all members of the committee have access.
+2. The Steering Committee can be reached publicly by opening an issue on [the governance repository](https://github.com/prometheus/governance/issues) and indicating that you would like attention from the steering committee.
```
Let's consistently capitalize Steering Committee (here and elsewhere).

RichiH replied:
Ignoring capitalization going forward; I agree

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+### Election Officers
+
+Six weeks or more before the election, the Steering Committee will appoint three Election Officer(s) to administer the election. Election Officers will be Organization Members in good standing who are eligible to vote, are not running for Steering Committee in that election, who are not currently part of the Steering Committee and can make a public promise of impartiality. 
```
```suggestion
Six weeks or more before the election, the Steering Committee will appoint three Election Officer(s) to administer the election. Election Officers will be Organization Members in good standing who are eligible to vote, are not running for Steering Committee in that election, who are not currently part of the Steering Committee, and can make a public promise of impartiality. 
```

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Anyone who is an Organization Member as defined in [ROLES.md](#roles.md).
+
+The electoral roll of all eligible voters will be captured at [elections](./elections/) and the voters’ guide will be captured at [voters guide](./elections/).
```
Still need to complete these links, right?

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+### Voting Procedure
+
+Elections will be held using a ranked choice voting system which supports voter privacy.  Specific voter tooling will be selected by the Election Officers before the election starts, with this selection to be approved by the Steering Committee. The top vote-getters will be elected to the open seats, with the exceptions for company representation discussed below.
```
The choice of the voting system actually matters. It might be infeasible to describe the voting system perfectly here, but then let's make it explicit that the current steering committee selects the exact voting system.

Even if we say here that we use Condorcet, in the implementation found on https://civs1.civs.us/ , there are still free parameters. For example, for a committee vote, using the "Proportional Representation" feature might be quite appropriate, which might have very tangible effect on the committee selection, see https://civs1.civs.us/proportional.html .

RichiH replied:
Proportional Representation was not a success in the Grafana hackathon voting. I'd use Condorcet with Schulze method for tie-breaking as SPI, Debian, etc are using that, and have been for decades.

beorn7 replied:
Are they electing multiple people with that?

Condorcet is meant to pick a single winner, not a group of winners. Hackathon is mostly about picking the winner, and then maybe about the ranking.

Electing the steering committee is not about a ranking. it is about finding the "correct" 7 people to be on the committee, with no ranking among them. Especially in our case where we don't want dominance of certain groups. proportional representation seems required.

I like https://en.wikipedia.org/wiki/Schulze_STV . (Not that this is different from "the Schulze method", which is https://en.wikipedia.org/wiki/Schulze_method .)

gouthamve replied:
Per the last dev-summit: https://docs.google.com/document/d/1uurQCi5iVufhYHGlBZ8mJMK_freDFKPG0iYBQqJ9fvA/edit?tab=t.0#bookmark=id.e6o3gm2zhv1z

we have decided to keep the current wording.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+### Limitations on Company Representation
+
+No more than 2 seats may be held by employees of the same organization (or  
+conglomerate, in the case of companies owning each other). If an election results in greater than 2 employees of the same organization being selected, the lowest vote getters from any particular employer will be removed until representation on the committee is down to 2.
```
Problem here is that it bakes in the assumption that the used election system spits out a ranking of candidates.

There are ranked choice voting systems that do not do that but just say "here are the 4 people elected for the committee". An example is the aforementioned CIVS with proportional representation. Or https://en.wikipedia.org/wiki/Schulze_STV. If you change the number of people to be elected or remove individual candidates from the result calculation (but whom in the first place?), results might come out quite differently.

So again, we either have to leave it open to the steering committee to flesh out the details for each election, or we have to describe a complete procedure.

Maybe, for starters, we should keep it simple and mandate a simple Condorcet-like system that just spits out a ranking, like vanilla https://civs1.civs.us/ without proportional representation. It's sub-optimal from certain perspectives, but I guess within the Prometheus project, that's way less problematic than in "real" political elections out there.

(Would be interesting to know what precise options CNCF usually uses for committee votes on https://civs1.civs.us/ . @RichiH , you might be the person that knows the details here.)

sysadmind replied:
Based on the wording, "the lowest vote getters from any particular employer", that implies that it could be either an existing member or a newly elected member (albeit silly to assume it could be an existing member). I think the wording should be clear that an election can not add steering committee members which would result in more than 2 employees from the same employer.

RichiH replied:
Any voting system can construct tied votes. In majority and first past the post voting systems, 4 people each receiving 5 votes will tie them if 3 slots are to be filled.

It seems unlikely that currently 51 people will vote in such a manner, in particular if they do not know other votes until completion and as such can only strategically vote to some extent.

I'd leave that open, re-vote, or if push comes to shove just add enough seats to SC to make sure everyone who is in such a tie makes it in.

beorn7 replied:
Ties ultimately have to be resolved by a random ballot, but the voting system should try to avoid ties (like the Schulze methods, but Schulze himself recommended a random bullet as the final tiebreaker).

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+## Changes to the Governance
+
+Changes to this Governance may be proposed via a Pull Request on the Governance itself.  
+Amendments are accepted with 2/3rd majority consent of the committee as per the [Decision Process](#decision-process) outlined above.
```
Again, this should state how to handle absences and abstains. I would propose a wording that requires two thirds of all current committee members to cast an approving vote.

RichiH replied:
Quorum & condorcet IMO. The supermajority equivalent for Condorcet is a higher quorum

beorn7 replied:
Condorcet only makes sense for a vote between more than two alternatives. I think governance changes are more individual yes/no decisions. Adding complications of Condorcet votes seem counter productive.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Ideally, most decisions will be made at the lowest possible level within the project: within individual working groups. When this is not possible, the Steering Committee can help facilitate a conversation to work through the contended issue. When facilitation by the Steering Committee does not resolve the contention, the Steering Committee may have to decide.
+
+For example, Technical governance is expected to be performed by the Maintainers of the various project areas and subprojects, as defined in the appropriate OWNERS / MAINTAINERS files.
```
Why is "Technical" capitalized?

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+* Consequences of being inactive include
+  * Involuntary removal or demotion
+  * Being asked to move to Emeritus status
+  * Not counting against quorum in votes
```
Are there any quorums in votes by members?

For the steering committee, this might actually be quite crucial in crunch situations. The removal from the vote should be more formalized than "inactivity". It is actually included below in the involuntary removal procedure for steering committee members, so maybe that's enough and we can delete the line here?

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Involuntary removal/demotion of a contributor happens when responsibilities and requirements aren't being met. This may include repeated patterns of inactivity, extended periods of inactivity, a period of failing to meet the requirements of your role, and/or a violation of the Code of Conduct. This process is important because it protects the community and its deliverables while also opening up o ...
+
+Inactive members can be automatically removed.
```
Since the definition of contributions is deliberately fuzzy (see ROLES.md), I find it hard to claim there is "automatic removal". As the steering committee decides what qualifies a contribution, the steering committee should also be the instance that determines inactivity.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Inactive members can be automatically removed.
+
+Involuntary removal or demotion of Maintainers or Steering Committee members is handled through a vote by a majority of the current Steering Committee. If this pertains to a Steering Committee member, that member is not eligible to discuss or vote on the matter.
```
This should include the impact on the quorum (more than half, in this case).

Speaking of "more than half": I find this procedure quite problematic. You effectively need "only" a majority decision to kick out a single member. This opens a path to use a simple majority to become a 2/3rd majority.

I mean, with only 7 members, the differences are not that much. Even if we required a 2/3rd majority to kick out a member, but that member cannot have their own vote in it, then 4 out of the remaining 6 is already a 2/3rd majority.

I don't think that inactivity of a steering committee member is a huge disaster as there always will be another vote in the future. So I would be more conservative here to avoid loopholes to "take over the project". Outcome should be that we need 5 approving votes to kick out a steering committee member (so effectively not removing the member to be removed from the quorum).

RichiH replied:
THey should be eligible to discuss, just not to vote

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+If and when contributors' commitment levels change, contributors can consider stepping down (moving down the contributor ladder) vs moving to emeritus status (completely stepping away from the project).
+
+Contact the Maintainers about changing to Emeritus status, or reducing your contributor level.
```
Why Maintainers (and not the steering committee)?

sysadmind replied:
I think the Steering Committee would only **need** to be involved when this is involuntary. Are there other situations where it makes sense for the Steering Committee to be involved?

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+For inquiries, please reach out to:  
+* [the governance repository](https://github.com/prometheus/governance/issues): Governance Repo Issues  
+* [prometheus-steering@googlegroups.com](mailto:prometheus-steering@googlegroups.com): Steering Committee Email
```
See above about `steering@prometheus.io`

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+
+## Contributor Ladder
+
+Hello! We are excited that you want to learn more about our project contributor ladder! This contributor ladder outlines the different contributor roles within the project, along with the responsibilities and privileges that come with them. Community members generally start at the first levels of the "ladder" and advance up it as their involvement in the project grows. Our project members are hap ...
```
I believe there is an implication that a "higher" level on the ladder includes the "lower" levels, e.g. if you are a Maintainer, you are also considered a Member. If you are a Member, you are also considered a Contributor. Maybe this should be stated explicitly.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Inactivity
+
+It is important for members to be and stay active to set an example and show commitment to the project. Inactivity is harmful to the project as it may lead to unexpected delays, contributor attrition, and a loss of trust in the project.
```
members → Members ?

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+All current Prometheus team members become members as per new governance. 
+
+The individual projects already define who is a maintainer for which specific project in their corresponding MAINTAINERS.md. People listed in these files are therefore going to be maintainers. 
```
Does this include all repos in github.com/prometheus-community, too?

Also, because Maintainers are also Members (I believe), we would make all Maintainers of all repos Members, which are a few (in case of github.com/prometheus) and a significant amount (in case of github.com/prometheus-community) new Members that aren't prometheus-team members right now.

Also note that some of the "auto-converted" people have been inactive for a while and would even qualify for immediate "auto-removal" if we left that part intact (see my comment above). Which would create needless pain and friction right in the beginning.

Personally, I don't think it's a huge problem to convert a bunch of inactive people, but we should clearly state what we intent here.

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+	* Authoring or reviewing PRs on GitHub
+	* Filing or commenting on issues on GitHub
+	* Organizing and running activities (e.g. events, surveys) within the Prometheus community
+	* Contributing to SIGs, subprojects, or community discussions (e.g. meetings, chat, email, and discussion forums)
```
This is another list of "qualifying contributions" – we already have one above in the "Contributor" section. Are the differences between the two lists deliberate? They do not feel much different. Maybe this should just say something like "See the Contributors section for qualifying contributions.") On the other hand, if it is intentional that some contributions that qualify for the Contributor are not qualifying for the Member role, call those differences out here explicitly.

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+
+1. Any current Maintainer may nominate a current Organization Member to become a new Maintainer, by opening a PR by adding them to the `MAINTAINERS.md` file.
+2. The nominee will add a comment to the PR testifying that they agree to all requirements of becoming a Maintainer.
+3. None of the other maintainers for the project must object.
```
Does "the project" imply "the whole Prometheus project"? Or just the repository where the MAINTAINERS.md file lives in?

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+
+Process of becoming a maintainer:
+
+1. Any current Maintainer may nominate a current Organization Member to become a new Maintainer, by opening a PR by adding them to the `MAINTAINERS.md` file.
```
Organization Member → Member

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+3. Once Sponsors have responded positively, the request will be reviewed by the Steering Committee or its delegates to make sure certain contributions are valid and non-trivial.
+4. If the threshold is met, the Contributor is added to [https://github.com/orgs/prometheus](https://github.com/orgs/prometheus)/people and [https://github.com/orgs/prometheus-community/people](https://github.com/orgs/prometheus-community/people) and becomes a Member
+
+### Maintainer
```
A general thought, assuming that the ladder is indeed "inclusive" (Maintainers are also Members): What happens to the maintainer of a tiny repo in prometheus-community that "just works" and doesn't need a lot of maintenance. They might cut a release or two a year, respond to the one and a half issues filed, and they are doing this super reliably. That's less than 20 contributions a year, and we are even talking about "auto-demotion" above.

I would like to see this case handled fairly. As already commented above, I don't think we should mention a notion of "auto-demotion", and maybe we can somehow mention that scenarios like the above exist, so that nobody will accuse the steering committee of not doing their job if they leave a maintainer like that in place.

beorn7 reviewed (changes requested):
Thanks for driving this.

I read through it and just collected some thought and notes. Sorry for the amount of things…

sysadmind commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+### Eligibility to Vote
+
+Anyone who is an Organization Member as defined in [ROLES.md](#roles.md).
```
To make a full sentence

```suggestion
Anyone who is an Organization Member as defined in [ROLES.md](#roles.md) may vote.
```

sysadmind commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+No more than 2 seats may be held by employees of the same organization (or  
+conglomerate, in the case of companies owning each other). If an election results in greater than 2 employees of the same organization being selected, the lowest vote getters from any particular employer will be removed until representation on the committee is down to 2.
+
+If employers change because of job changes, acquisitions, or other events, in a way that would yield too many seats being held by employees of the same organization, and the next regular election is more than 10 weeks away, sufficient members of the committee must resign until only 2 employees of the same employer are left. If it is impossible to find sufficient members to resign, all employees o ...
```
I think this should consider if none of the employee seats are up for election that cycle. To that end, what if the employee which resigns is from the opposite election cycle? Would that lead to a 1 year term, or an extra seat in the current cycle?

RichiH replied:
```suggestion
If affiliations change because of job changes, acquisitions, or other events and would result in more than 2 affiliated individuals serving on the Steering Committee and the next regular election is more than 10 weeks away, a sufficient number of members of the Steering Committee must resign. If this does not happen within two weeks, all affiliated individuals from such a set are removed and backfilled through normal backfill mechanisms.
```

sysadmind commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Stepping Down/Emeritus Process
+
+If and when contributors' commitment levels change, contributors can consider stepping down (moving down the contributor ladder) vs moving to emeritus status (completely stepping away from the project).
```
This lower section seems to talk about the roles beyond the Steering Committee. For me, reading this was confusing because most of this document is focused solely on the Steering Committee. I think this section might make more sense with the roles instead of here.

sysadmind commented on ROLES.md:
```diff
@@ -0,0 +1,99 @@
+  * Mentors other contributors
+* Additional privileges
+  * Represent the project in public as a Maintainer
+  * Have a vote in Maintainer decision-making meetings
```
One thing not defined here is access to sensitive information like CI/CD secrets, passwords, etc that a repo might need for CI/CD, docker image publishing, etc, and who manages those sensitive items. I think it would be good to spell that out.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+The individual projects already define who is a maintainer for which specific project in their corresponding MAINTAINERS.md. People listed in these files are therefore going to be maintainers. 
+
+The existing Prometheus team is going to hold votes, as described in the new governance, to elect a Bootstrap steering committee of 7 members, where the highest voted 3 members will hold a 2 year and the next 4 members will hold a 1-year term.  
```
If we make include "maintainers but not team members" as initial Members according to the new governance, should we also let them take part in the initial steering committee election?

gouthamve replied:
Yes.

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Values
+
+The Prometheus developers and community are expected to follow the values defined in the [CNCF charter](https://www.cncf.io/about/charter/), including the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). Furthermore, the Prometheus community strives for kindness, giving feedback effectively, and building a welcoming environment. The Prometheus developers  ...
```
```suggestion
The Prometheus developers and community are expected to follow the values defined in the [CNCF charter](https://www.cncf.io/about/charter/), including the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). Furthermore, the Prometheus community strives for kindness, giving feedback effectively, and building a welcoming environment. The Prometheus developers default to deciding by consensus and only fall back to resolution by voting if consensus cannot be reached.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Charter
+
+The following responsibilities and powers belong to the Steering Committee.
```
```suggestion
The Steering Committee has the following responsibilities and powers.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+* Define, evolve, and promote the vision, values, and mission of the project.
+* Define and evolve project governance structures and policies, including project roles and how collaborators become members, approvers, leads, and/or administrators. This includes policy for the creation and administration of working groups and committees.
+* Steward, control access, delegate access, and establish processes regarding all Prometheus project resources and has the final say in the disposition of those resources.
```
```suggestion
* Steward, control access, delegate access, and establish processes regarding all Prometheus project resources, having the final say in the disposition of those resources.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+* Define, evolve, and promote the vision, values, and mission of the project.
+* Define and evolve project governance structures and policies, including project roles and how collaborators become members, approvers, leads, and/or administrators. This includes policy for the creation and administration of working groups and committees.
+* Steward, control access, delegate access, and establish processes regarding all Prometheus project resources and has the final say in the disposition of those resources.
+* Coordinate with the CNCF regarding usage of the Prometheus brand and how that brand can be used in relation to other efforts or vendors.
```
```suggestion
* Coordinate with the LF and the CNCF regarding usage of the Prometheus trademark and how that trademark can be used in relation to and with other efforts, projects, or vendors.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+* Define and evolve project governance structures and policies, including project roles and how collaborators become members, approvers, leads, and/or administrators. This includes policy for the creation and administration of working groups and committees.
+* Steward, control access, delegate access, and establish processes regarding all Prometheus project resources and has the final say in the disposition of those resources.
+* Coordinate with the CNCF regarding usage of the Prometheus brand and how that brand can be used in relation to other efforts or vendors.
+* Receive and handle reports about code of conduct violations and maintain confidentiality.
```
```suggestion
* Receive and handle reports about Code of Conduct violations while maintaining confidentiality.
```

RichiH replied:
Problem: What if there's a CoC complaint against a sitting SC member? Yeet it to CNCF CoCC, presumably? Should we codify that?

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Given the private nature of some of these discussions (e.g. privacy, private emails to the committee, code of conduct violations, escalations, disputes between members, security reports, etc.) some meetings are held in private.
+
+Public meeting recordings and notes are public and private meeting notes are private.
```
```suggestion
```

It seems self-evident?

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Committee members
+
+Seats on the Steering Committee are held by an individual, not by their employer.
```
```suggestion
Seats on the Steering Committee are held by individuals, not by or through their respective employers.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Seats on the Steering Committee are held by an individual, not by their employer.
+
+The current membership of the committee is currently (listed alphabetically by first name):
```
```suggestion
The current membership of the Steering Committee, listed alphabetically by first name:
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+Decisions requiring a vote include: issuing written policy, amending existing  
+written policy, creating, removing, or modifying a working group, all spending,  
+hiring, and contracting, official responses to publicly raised issues, or any  
+other decisions that at least two of the members present decide require a vote.
```
```suggestion
other decisions that at least two of the Steering Committee members present ask for a vote for.
```

"decide" implies too much power IMO; asking for a vote is enough

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+# Prometheus Governance 2.0
```
High level comment: Goutham and myself decided to skip ~everything related to voting as this needs a wider discussion.

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Getting in touch
+
+There are two ways to raise issues to the steering committee for decision:
```
```suggestion
There are two ways to raise issues to the Steering Committee for decision:
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Composition
+
+The steering committee has 7 seats. These seats are open to any project member. See [Candidate Eligibility](#candidate-eligibility) for a definition.
```
```suggestion
The Steering Committee has 5 seats. These seats are open to any project member. See [Candidate Eligibility](#candidate-eligibility) for a definition.
```

At 51 members, ~1/7th seems large; ~1/10th seems better.

beorn7 replied:
I would also prefer a smaller steering committee, but not for this reason.

I expect the project to have _way_ more than 51 members very soon. There are 170 individuals with 20+ contributions over the last year (according to CNCF dev-stats).

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+The steering committee has 7 seats. These seats are open to any project member. See [Candidate Eligibility](#candidate-eligibility) for a definition.
+
+Steering Committee members serve for 2 year terms, staggered to preserve continuity. Every year, either 4 or 3 contributor seats are elected.
```
```suggestion
Steering Committee members serve for 2 year terms, staggered to preserve continuity. Not accounting for backfill, this means that either 3 or 4 Steering Committee members are elected per year.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+### Timeline
+
+Steering Committee elections are held annually. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).  
+Four weeks or more before the election, the Election Officer(s) will issue a call for nominations, publish the list of voters. 
```
```suggestion
Four weeks or more before the election, the Election Officer(s) will issue a call for nominations and publish the list of eligible voters. 
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+Steering Committee elections are held annually. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).  
+Four weeks or more before the election, the Election Officer(s) will issue a call for nominations, publish the list of voters. 
+
+Seven days before the election, the call for nominations will be closed. The election will be open for voting for no less than two weeks and no more than four. The results of the election will be announced within one week of closing the election. New Steering Committee members will take office is May or June of each year.
```
```suggestion
Seven days before the election, the call for nominations will be closed. The election will be open for voting for no less than two weeks and no more than four weeks. The result of the election will be announced within one week of closing the election.
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+### Timeline
+
+Steering Committee elections are held annually. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).  
```
```suggestion
Steering Committee elections are held annually, aiming to seat new members in June. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).  
```

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+### Candidate Eligibility
+
+Community members must be eligible to vote to stand for election. Candidates should self-nominate. There are no term limits for Steering Committee members.
```
```suggestion
Organization members must be eligible to vote to stand for election. Candidates should self-nominate. There are no term limits for Steering Committee members.
```

Though we should probably say "project" not "org" members throughout?

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+### Limitations on Company Representation
+
+No more than 2 seats may be held by employees of the same organization (or  
+conglomerate, in the case of companies owning each other). If an election results in greater than 2 employees of the same organization being selected, the lowest vote getters from any particular employer will be removed until representation on the committee is down to 2.
```
```suggestion
No more than 2 Steering Committee seats may be held by individuals with the same professional affiliation, in particular employees of, contractors receiving a majority of their compensation from, or significant shareholders/investors of the same organization or conglomerate. Affected individuals from such a set with more recent and/or more total votes will be given preference, though anyone from this set can recuse themselves for selection until the maximum of 2 has been reached.
```

metalmatze replied:
@gouthamve and I have been discussing the "compensation from, or significant shareholders/investors" part quite a bit in the call just now. 

Could you simplify the statement? Currently it could be read as three employees from two different companies with the same investor not being allowed to be on the Steering Committee.

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+If employers change because of job changes, acquisitions, or other events, in a way that would yield too many seats being held by employees of the same organization, and the next regular election is more than 10 weeks away, sufficient members of the committee must resign until only 2 employees of the same employer are left. If it is impossible to find sufficient members to resign, all employees o ...
+
+In the event of a question of company membership (for example evaluating independence of corporate subsidiaries) a majority of all non-involved Steering Committee members will decide.
```
NB: Condorcet

beorn7 replied:
Really? That would only matter if they have to vote between more than two options for company affiliation…

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+In the event of a question of company membership (for example evaluating independence of corporate subsidiaries) a majority of all non-involved Steering Committee members will decide.
+
+## Vacancies
```
Rename everything here to backfill?

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Changes to the Governance
+
+Changes to this Governance may be proposed via a Pull Request on the Governance itself.  
```
only place governance and PR are capitalized. I'm fine either way, just noting

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+Changes to this Governance may be proposed via a Pull Request on the Governance itself.  
+Amendments are accepted with 2/3rd majority consent of the committee as per the [Decision Process](#decision-process) outlined above.
+
+Proposals and amendments to the charter are available for at least a period of one week for comments and questions before a vote will occur.
```
Two weeks?

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Authority, Facilitation, and Decision Making
+
+Ideally, most decisions will be made at the lowest possible level within the project: within individual working groups. When this is not possible, the Steering Committee can help facilitate a conversation to work through the contended issue. When facilitation by the Steering Committee does not resolve the contention, the Steering Committee may have to decide.
```
I'd put the SC later. Let things bubble up to e.g. dev summit without calling on SC for everything

metalmatze replied:
@gouthamve and I conclude that deciding "within individual working groups" includes dev summits and we don't have to explicitly state where decisions have to be made until SC is brought in.

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+For example, Technical governance is expected to be performed by the Maintainers of the various project areas and subprojects, as defined in the appropriate OWNERS / MAINTAINERS files.
+
+Note that if the committee is called to resolve contended decisions regularly, it is a symptom of a larger problem in the community that will need to be addressed.
```
```suggestion
Note that if the Steering Committee is called to resolve contended decisions regularly, it is a symptom of a larger problem in the community that will need to be addressed.
```

I am not sure this is always true. We currently don't have voting for all members other than voting for SC? Just delete this sentence?

metalmatze replied:
What do you mean by this comment? 

The sentence doesn't mention voting or anyway to resolve the problem but only states that constantly having the SC to be called in to resolve problems is something that should be looked at.

RichiH commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+## Involuntary Removal or Demotion
+
+Involuntary removal/demotion of a contributor happens when responsibilities and requirements aren't being met. This may include repeated patterns of inactivity, extended periods of inactivity, a period of failing to meet the requirements of your role, and/or a violation of the Code of Conduct. This process is important because it protects the community and its deliverables while also opening up o ...
```
Stray "you", though I feel as if a list of bullet points is better here?

ArthurSens commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,179 @@
+
+Decisions requiring a vote include: issuing written policy, amending existing  
+written policy, creating, removing, or modifying a working group, all spending,  
+hiring, and contracting, official responses to publicly raised issues, or any  
```
Can we even do hiring and contracting?

siavashs committed:
fix: roles link

Fix governance.md link to roles.md

Signed-off-by: Siavash Safi <siavash@cloudflare.com>

gouthamve committed:
Merge pull request #3 from siavashs/fix/roles-link

fix: roles link

gouthamve committed:
Address review feedback

Thanks for all the reviews Beorn, Richi and Joe!

Signed-off-by: Goutham <gouthamve@gmail.com>

metalmatze commented:
@gouthamve and I have worked on all comments and resolved them only to realize GitHub only showed us 40 of them. We're going to meet next week Fri again to work on the remaining comments.

metalmatze committed:
Improve wording and clarify governance rules

- Reword charter responsibilities for clarity
- Update trademark language to reference LF and CNCF
- Rename "Vacancies" to "Backfilling"
- Clarify affiliation change rules for SC members
- Extend amendment comment period to two weeks
- Various grammar and style improvements
- Add TODOs for Code of Conduct violation handling and election timing

Signed-off-by: Matthias Loibl <mail@matthiasloibl.com>

metalmatze committed:
Grammarly pass with humans.

Signed-off-by: Matthias Loibl <mail@matthiasloibl.com>

ArthurSens commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,183 @@
+* Define and evolve project governance structures and policies, including project roles and how collaborators adopt these roles. This includes policy for the creation and administration of working groups and committees.
+* Steward, control access, delegate access, and establish processes regarding all Prometheus project resources, having the final say in the disposition of those resources.
+* Coordinate with the LF and the CNCF regarding usage of the Prometheus trademark and how that trademark can be used in relation to and with other efforts, projects, or vendors.
+__TODO__: Discuss what happens if SC members is violating CoC.
```
Should this be escalated to LF? Or may the reporter reach out to the remaining SC member privately so they can address the matter?

bboreham replied:
I had the same reaction, escalate to CNCF.

https://www.kubernetes.dev/community/code-of-conduct-incident-process/ says
> If all members of the Code of Conduct Committee felt the need to recuse themselves from an incident, the incident would be handled by our third party mediator.

bboreham commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,183 @@
+
+To pass a decision, it needs approving votes from more than half of the current Steering Committee members.
+
+__TODO__: Explain how to handle multi-proposal votes.
```
Ban them?

gouthamve replied:
Well, kinda. Only `yes/no` votes, and any multi-proposal vote means each proposal gets it's own vote.

bboreham commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,183 @@
+
+### Timeline
+
+Steering Committee elections are held annually, aiming to seat new members in __TODO__. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).
```
"aiming to seat new members in 1 month"?

bboreham commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,183 @@
+  * Periods of no contributions for longer than 6 months
+  * Not having 20 contributions in the preceding 12-month window for more than 6 months
+
+* Consequences of being inactive include
```
"may include" ?  Otherwise we are committing to administrative work that may have little benefit.

bboreham commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,183 @@
+
+There are two ways to raise issues to the Steering Committee for decision:
+
+__TODO__: Figure out if we want to keep the archive or a mailing-alias.
```
I didn't understand the issue here.

bboreham commented on ROLES.md:
```diff
@@ -0,0 +1,105 @@
+  * Contribute to the documentation
+  * Participate in dev summits or workgroup meetings
+  * Answer questions from other community members
+  * Help test releases
```
s/Help to test new releases/  I sat there thinking "what do they mean by a test release" ?

bboreham commented on ROLES.md:
```diff
@@ -0,0 +1,105 @@
+
+## Stepping Down/Emeritus Process
+
+If and when contributors' commitment levels change, contributors can consider stepping down (moving down the contributor ladder) vs moving to emeritus status (completely stepping away from the project).
```
I don't understand "emeritus" as meaning "completely stepping away", more that it is a mark of esteem; that person was particularly valued in the role.  
Wikipedia says "The term emeritus does not necessarily signify that a person has relinquished all the duties of their former position, and they may continue to exercise some of them"

bboreham commented on ROLES.md:
```diff
@@ -0,0 +1,105 @@
+* Requirements
+  * Consistently contributing to the codebase
+  * Reviewer for or author of substantial PRs to the codebase, with the definition of substantial subject to the maintainer's discretion (e.g. refactors/adds new functionality rather than one-line pulls).
+  * Demonstrates a broad knowledge of the project across multiple areas
```
Is this helpful?  Would we really refuse someone because all they knew was Kubernetes service discovery?

bboreham commented on ROLES.md:
```diff
@@ -0,0 +1,105 @@
+  * Reviewer for or author of substantial PRs to the codebase, with the definition of substantial subject to the maintainer's discretion (e.g. refactors/adds new functionality rather than one-line pulls).
+  * Demonstrates a broad knowledge of the project across multiple areas
+  * Can exercise judgment for the good of the project, independent of their employer, friends, or team
+  * Mentors other contributors
```
Again, do we want to hold up someone being a maintainer until they have found someone they can mentor?

bboreham commented on ROLES.md:
```diff
@@ -0,0 +1,105 @@
+A Maintainer must meet the responsibilities and requirements of an Organization Member, plus:
+
+* Responsibilities include
+  * Demonstrate sound technical judgment (maybe asked to step down by a maintainer if they lose the confidence of the other maintainers)
```
"step down as a maintainer" ?
I think this parenthetical can be covered in the "Involuntary Removal or Demotion" section, and we don't need it here.

bboreham reviewed (commented):
Great stuff!  Some thoughts below.

gouthamve committed:
Handle final TODOs

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

gouthamve committed:
Address Bryan's feedback

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

codesome commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+1. Open an GitHub Issue against the [**TODO:** prometheus/community repo]
+   1. Ensure your sponsors are @mentioned on the issue
+   2. Assign label **TODO**
+   3. Complete every item on the checklist (**TODO:** template)
```
Will these TODOs be updated before merging?

beorn7 replied:
Yeah, I don't think we should have a TODO in a text that is going for a formal vote.

codesome commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,181 @@
+
+## Authority, Facilitation, and Decision-Making
+
+Ideally, most decisions will be made at the lowest possible level within the project: within individual working groups. When this is not possible, the Steering Committee can help facilitate a conversation to work through the contested issue. When the Steering Committee's facilitation does not resolve the contention, the Steering Committee may have to decide.
```
Making the authority a little stronger here. `'may have to decide'` sounds like there can be situations where the contention stays and the decision is not taken.

```suggestion
Ideally, most decisions will be made at the lowest possible level within the project: within individual working groups. When this is not possible, the Steering Committee can help facilitate a conversation to work through the contested issue. When the Steering Committee's facilitation does not resolve the contention, the Steering Committee will take the final decision.
```

gouthamve replied:
> sounds like there can be situations where the contention stays and the decision is not taken.

Yes, this can be the case too :) I can see cases where the steering committee can put a contentious decision on hold.

codesome commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,181 @@
+
+## Contact
+
+For inquiries, please reach out to:  
```
It is not clear where would I reach out for what purpose between the two options below. Should we make it more explicit?

codesome commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,181 @@
+
+For inquiries, please reach out to:  
+* [the governance repository](https://github.com/prometheus/governance/issues): Governance Repo Issues  
+* [prometheus-steering@googlegroups.com](mailto:prometheus-steering@googlegroups.com): Steering Committee Email
```
It reads better this way
```suggestion
* Governance Repo Issues: https://github.com/prometheus/governance/issues
* Steering Committee Email: [prometheus-steering@googlegroups.com](mailto:prometheus-steering@googlegroups.com)
```

codesome commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,181 @@
+
+The Steering Committee has 7 seats. These seats are open to any project member. See [Candidate Eligibility](#candidate-eligibility) for a definition.
+
+Steering Committee members serve for 2-year terms, staggered to preserve continuity. Not accounting for backfill, this means that either 3 or 4 Steering Committee members are elected per year.
```
This line was a little confusing until I read the migration section where it said

> 3 highest-voted members serving 2-year terms and the next 4 serving 1-year terms

We should make that explicit in this section as well.

codesome replied:
~Also, can one be a committee member for consecutive terms or should there be a gap? I am personally on the side of allowing consecutive terms. Either way, better to make that explicit here.~

beorn7 replied:
It's actually stated explicitly below: "There are no term limits for Steering Committee members."

codesome reviewed (approved):
Looks great. I have some non-blocking comments, so approving it.

aknuds1 reviewed (approved):
LGTM

gouthamve committed:
More review feedback

* Use condorcet as decided in the dev summit
* Make the contact mechanism use explicit

Signed-off-by: Goutham <gouthamve@gmail.com>

SuperQ commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+## Roles
+
+We have a contributor ladder that includes the roles of Contributor, Member, Maintainer, and Steering Committee. The Steering Committee responsibilities and elections are outlined below. The remaining roles are detailed in [ROLES.md](./ROLES.md).
```
There is no `Steering Committee` section in ROLES.md.

beorn7 replied:
Since the steering committee is explained in detail here in GOVERNANCE.md, I wouldn't duplicate the explanations, but at least ROLES.md should contain a "Steering Committee" section that has a very short summary of what that is and then refers to the relevant section of the governance for detailed explanations.

Also of note, maybe, that a member can be elected onto the steering committee without being a maintainer, i.e. steering committee is not a "next step" on a linear ladder.

bwplotka replied:
I find this very confusing that we explain some roles in ROLES.md, some roles in this document. 

What about:
* Document all roles in `GOVERNANCE.md`
* Have a separate `Ladder.md` document for describing typical flow?

Alternatively:
* Move stering commitee to ROLES.md and remove from here.

ArthurSens replied:
See https://github.com/prometheus/governance/pull/1/changes#r2919677838

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+This document describes the rules and governance of the Prometheus project. It is meant to be followed by all the developers of the project and the Prometheus community. This governance is an open, living document, and will continue to evolve as the community and project change.
+
+It outlines the Steering Committee's roles and responsibilities, the decision-making process, and the election procedures for committee members. It also details the contributor ladder, including the roles of Contributor, Member, and Maintainer, along with the requirements and privileges associated with each.
```
As the ladder includes "Steering Committee" in line 15 below, I'd include it here as well for consistency.

bwplotka replied:
I'd also suggest linking where to find description of those roles exactly. Especially, given the current state has some roles define in this file, some in ROLES.md

ArthurSens replied:
I believe a contribution ladder is inspired by a career ladder, which is usually present in organizations, where there is a set of clear requirements that an individual is assessed against, and a clear outcome: PASS/FAIL. 

Becoming a Steering Committee member does not work like this. There are no clear requirements for joining the Steering Committee; it's election-based, so assessing requirements against a given individual's contributions doesn't make sense.

I'll make this a bit clearer in the document.

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+### Voting Procedure
+
+Elections will be held using the condorcet voting method. Specific voter tooling will be selected by the Election Officers before the election starts, with this selection to be approved by the Steering Committee. The top vote-getters will be elected to the open seats, with the exceptions for company representation discussed below.
```
Two nits:
1. Capitalize "Condorcet". It's a person.
2. Make explicit that we need a ranking method. Vanilla Condorcet just finds a single winner, see https://en.wikipedia.org/wiki/Condorcet_method#Condorcet_ranking_methods


```suggestion
Elections will be held using a ranking Condorcet voting method picked by the Election Officers. Specific voter tooling will be selected by the Election Officers before the election starts, with this selection to be approved by the Steering Committee. The top vote-getters will be elected to the open seats, with the exceptions for company representation discussed below.
```

beorn7 commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+### Limitations on Company Representation
+
+No more than 2 seats may be held by employees of the same organization (or  
```
Still a bit unhappy with the use of "employee". In our modern world of remote work, many "employees" are formally just contractors. I think what we mean here are people who earn the majority of their income from a single source. Maybe there is an easy way to generalize this here.

bwplotka replied:
Majority of income would be ok, except, how would you prove it in a failure mode though 🙃 

What about:

```suggestion
No more than 2 seats may be held by employees or contractor of the same organization (or  
```

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+* Requirements
+  * Must meet the requirements of a Contributor
+	* Must have two or more Maintainers as sponsors. At least one sponsor must have a different employer than the applying Contributor
+	* Must enable 2FA on their GitHub account
```
Inconsistent indentation.

```suggestion
  * Must have two or more Maintainers as sponsors. At least one sponsor must have a different employer than the applying Contributor
  * Must enable 2FA on their GitHub account
```

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+* Responsibilities include
+  * Continuing to contribute regularly.
+	* This can be demonstrated by having at least 20 GitHub contributions a year in [DevStats](https://devstats.cncf.io/).
+	* We encourage non-code contributions like attending and contributing to dev summit and workgroup meetings, organizing and running activities within the Prometheus community, etc.
```
Inconsistent indentation (tabs vs. spaces)

```suggestion
  * This can be demonstrated by having at least 20 GitHub contributions a year in [DevStats](https://devstats.cncf.io/).
  * We encourage non-code contributions like attending and contributing to dev summit and workgroup meetings, organizing and running activities within the Prometheus community, etc.
```

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+	* We encourage non-code contributions like attending and contributing to dev summit and workgroup meetings, organizing and running activities within the Prometheus community, etc.
+
+* Requirements
+  * Must meet the requirements of a Contributor
```
It appears you could become a Member with fewer than 20 contributions as long as there is a realistic expectations you will do at least 20 contributions a year once you are a Member.

Is that intended?

If not, might it be clearer to add the contribution level (20 contributions / year) here in the requirements section and then have the responsibilities section say that this level has to be maintained?

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+  * Mentoring new contributors
+  * Determining strategy and policy for the project
+  * Participating in, and leading, community meetings
+  * May merge PRs
```
```suggestion
  * May merge PRs affecting the code they are Maintainers for
```

Additional technicality: There might be PRs touching sub-trees with different maintainers, and then nobody can technically merge that PR. So maybe say "approve" instead of "merge"?

bwplotka replied:
Who will merge then? 

If different maintainers are in the path, this means we need approval of those first, no? Then the maintainer can merge?

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+  * Can exercise judgment for the good of the project, independent of their employer, friends, or team
+* Additional privileges
+  * Represent the project in public as a Maintainer
+  * Have a vote in Maintainer decision-making meetings
```
This implies that
1. Maintainers will vote about decisions
2. decisions are made in meetings

Which I think is both misleading. Maybe that's not the intention, but I understand the whole thing such that we will avoid any votes, and if a vote is required, it will be only within the steering committee. If we expect votes among Maintainers, that should be clarified somewhere. On the other hand, maybe it should be clarified that the only votes happening outside of electing the steering committee happens within the steering committee (and only as a last resort).

Concretely, I would word this paragraph as:

```suggestion
  * Take part in Maintainer decision-making
```

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+  * Represent the project in public as a Maintainer
+  * Have a vote in Maintainer decision-making meetings
+
+Process of becoming a maintainer:
```
Shouldn't the Steering Committee give final approval of any new Maintainer?

Also we might want to add a "bootstrapping" paragraph that the steering committee can appoint Maintainers for unmaintained parts of the codebase. (Can easily happen if the single maintainer of a repo disappears.)

sysadmind replied:
I think it would be reasonable to have the Steering Committee approve, but there should be an expectation of timeframe (i.e. final approval will happen at the next Steering Committee meeting)

gouthamve replied:
I don't think this will scale tbh. I am not sure we should gate this behind Steering committee approval.

I think if there is any _disagreement_, then we can then refer it to the committee. I agree with bootstrapping, though. Will add a section.

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+
+Process of becoming a maintainer:
+
+1. Any current Maintainer may nominate a current Member to become a new Maintainer, by opening a PR by adding them to the `MAINTAINERS.md` file.
```
Is this meant "in scope", i.e. only Maintainers of a certain repo (or sub-tree) can nominate new Maintainers for that repo (or sub-tree)? Or can any Maintainer nominate new Maintainers for any parts of the codebase?

beorn7 commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+
+1. Any current Maintainer may nominate a current Member to become a new Maintainer, by opening a PR by adding them to the `MAINTAINERS.md` file.
+2. The nominee will add a comment to the PR testifying that they agree to all requirements of becoming a Maintainer.
+3. None of the other maintainers of the repository must object.
```
```suggestion
3. None of the other maintainers of the relevant repository or part of the code base must object.
```

beorn7 reviewed (approved):
Bunch of minor comments. I don't see any as blocking, so I also approve. I would prefer resolving comments first, but since the steering committee can do this cleanup easily, it's not a hard requirement.

sysadmind commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+They will be responsible for:
+
+* Making all announcements associated with the election
```
Where would we expect these announcements to be made? If this would include social media, who is in control of those accounts in this new governance and how might they be granted that access? It would be good to explicitly call out where these announcements will be made.

gouthamve replied:
I think the steering committee or election officers will make this decision. I think it's fair to expect this announcement in `prometheus-developers@` and `prometheus-users@` mailing lists, with everything else being optional. Trying to list every single thing here would be futile as new channels are added and removed.

sysadmind commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+### Eligibility to Vote
+
+Anyone who is an Organization Member as defined in [ROLES.md](./ROLES.md) may vote.
```
This does not explicitly include the Maintainer role. Are Maintainers eligible to vote? I think based on the statement at the top of ROLES.md they are, but it's easy for this fact to be lost.  
```
Being in a "higher" level on the ladder includes the "lower" levels, e.g. if you are a Maintainer, you are also considered a Member. If you are a Member, you are also considered a Contributor and have the privileges and responsibilities of all the lower levels.
```

gouthamve replied:
Maintainers are members, you cannot be a maintainer without being a member. I am not sure we should over specify things in the governance doc, else it will become tedious.

ArthurSens replied:
This is actually written in the ROLES.md file

sysadmind commented on ROLES.md:
```diff
@@ -0,0 +1,103 @@
+  * Can recommend other contributors to become Org Members
+  * May be assigned Issues and Reviews
+  * May give commands to CI/CD automation
+  * Can manage CI and package registry secrets
```
It seems weird to me that a Member may manage CI and registry secrets. The responsibility to merge a PR shows up in Maintainer and I don't see managing CI for a project to be materially different than managing the code itself. I recommend that this item gets moved to the Maintainer role.

jan--f commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+### Candidate Eligibility
+
+Organization members must be eligible to vote to stand for election. Candidates should self-nominate. There are no term limits for Steering Committee members.
```
Does `Candidates should self-nominate.` mean that other can also nominate candidates?
Might be good to be more explicit here? Should self-nominate or/and can/must not be nomintes by other org members?

bwplotka replied:
Yea, I'd suggest indeed making it a MUST? It would be weird if someone nominate X and X does not want to participate (:

bwplotka replied:
Another suggestion -- can we link  `#limitations-on-company-representation` here? It's a relevant bit for eligibility.

gouthamve replied:
I don't think it's very relevant tbh, because just because 10 people from the same company participate, doesn't mean they'll all win. I'd rather all 10 interested parties run for the election.

Candidacy is explicitly linked to a person, and in Grafana we explicitly avoid coordinating who runs and who doesn't, because an interested individual shouldn't be discouraged from running because someone else in the same company is.

jan--f commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+Members need to be and stay active to set an example and show commitment to the project. Inactivity is harmful to the project, as it may lead to unexpected delays, contributor attrition, and a loss of trust.
+
+* Inactivity is measured by
```
Perhaps explicitly link to ROLES.md for definitions of contribution?

bwplotka commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+## Values
+
+The Prometheus developers and community are expected to follow the values defined in the [CNCF charter](https://www.cncf.io/about/charter/), including the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). Furthermore, the Prometheus community strives for kindness, effective feedback, and a welcoming environment. The Prometheus developers default to decisio ...
```
Voting vs consensus is important, I'd propose highlighting it by e.g. having it in another paragraph.

```suggestion
The Prometheus developers and community are expected to follow the values defined in the [CNCF charter](https://www.cncf.io/about/charter/), including the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). Furthermore, the Prometheus community strives for kindness, effective feedback, and a welcoming environment. 

The Prometheus developers default to decision by consensus and only fall back to voting if consensus cannot be reached.
```

bwplotka commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+Steering Committee elections are held annually, aiming to seat new members in March. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).
+Four weeks or more before the election, the Election Officer(s) will issue a call for nominations and publish the list of eligible voters.
+
+Seven days before the election, the call for nominations will be closed. The election will be open for voting for no less than two weeks and no more than four weeks. The election results will be announced within one week of the election closing.
```
What if we can't find a new member?

gouthamve replied:
Honestly, that's then something for the steering committee to step in a remediate. It's hard to catch every single edge-case here.

bwplotka reviewed (approved):
LGTM in general, but left some minor comments and question.

Especially the maintainer role and process of adding new maintainers has some gaps e.g. https://github.com/prometheus/governance/pull/1/files?diff=unified&w=0#r2736795551

gouthamve committed:
Apply suggestions from code review

Co-authored-by: Björn Rabenstein <github@rabenste.in>
Co-authored-by: Bartlomiej Plotka <bwplotka@gmail.com>
Signed-off-by: Goutham Veeramachaneni <gouthamve+github@gmail.com>
Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

gouthamve committed:
More review feedback

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

ArthurSens committed:
Grammatical fixes

Signed-off-by: Arthur Silva Sens <arthursens2005@gmail.com>

ArthurSens committed:
Explicitly say that Governance Committee is not part of contribution ladder

Signed-off-by: Arthur Silva Sens <arthursens2005@gmail.com>

ArthurSens committed:
Link contributions to its definition

Signed-off-by: Arthur Silva Sens <arthursens2005@gmail.com>

ArthurSens committed:
Clarify maintainer rejection

Signed-off-by: Arthur Silva Sens <arthursens2005@gmail.com>

ArthurSens committed:
More grammar fixes

Signed-off-by: Arthur Silva Sens <arthursens2005@gmail.com>

ArthurSens commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+Anyone who is an Organization Member as defined in [ROLES.md](./ROLES.md) may vote.
+
+The electoral roll of all eligible voters will be captured at [elections](./elections/), and the voters’ guide will be captured at the [voters' guide](./elections/).
```
This elections folder doesn't exist @gouthamve, is this supposed to be created during the election period?

gouthamve replied:
Yes, it will be created for the first election.

jan--f commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+
+### Timeline
+
+Steering Committee elections are held annually, with new members seated in March. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).
```
nit
```suggestion
Steering Committee elections are held annually, with new members seated in March. Six weeks or more before the election, the Steering Committee will appoint [Election Officers](#election-officers).
```

jan--f commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,182 @@
+### Timeline
+
+Steering Committee elections are held annually, with new members seated in March. Six weeks or more before the election, the Steering Committee will appoint [Election Officer(s)](#election-officers).
+Four weeks or more before the election, the Election Officer(s) will issue a call for nominations and publish the list of eligible voters.
```
```suggestion
Four weeks or more before the election, the Election Officers will issue a call for nominations and publish the list of eligible voters.
```

jan--f reviewed (approved):
Looks great!
A small nit: reads like there are always three election officers so no need to write `election officer(s)`, but feel free to ignore. Works as is.

ArthurSens committed:
Apply suggestions from code review

Co-authored-by: Jan Fajerski <jan--f@users.noreply.github.com>
Signed-off-by: Arthur Silva Sens <arthursens2005@gmail.com>

gouthamve commented:
Thanks everyone for the detailed feedback!!! It's GO TIME!
